lib/dialects/abstract/connection-manager.js

```diff
@@ -25,21 +32,19 @@ class ConnectionManager {
     this.config = config;
     this.dialect = dialect;
     this.versionPromise = null;
-    this.poolError = null;
     this.dialectName = this.sequelize.options.dialect;
 
     if (config.pool === false) {
       throw new Error('Support for pool:false was removed in v4.0');
     }
 
-    config.pool =_.defaults(config.pool || {}, defaultPoolingConfig, {
+    this.config.pool =_.defaults(config.pool || {}, defaultPoolingConfig, {
```

Comment:
Woosh, user supplied `pooling.config` was never used

Reply:
nit: space after equals

Reply:
While I agree with the change because it's clearer, doesn't `this.config === config`? And wouldn't that mean that changing `config` also changes `this.config`?

Reply:
`const config` is deep cloned version of `this.config`, so any changes made to it will not reflect into `this.config`